R/statistic-covariance.R

```diff
@@ -120,25 +120,30 @@ dpCovariance <- setRefClass(
 
 dpCovariance$methods(
   initialize = function(mechanism, varType, n, epsilon, columns, rng=NULL, imputeRng=NULL, 
-                        intercept=FALSE, formula=NULL, delta=1e-5) {
+                        intercept=FALSE, formula=NULL) {
+    # NOTE THIS IS CURRENTLY NOTE SECURE DUE TO SENS AND EPSILON NOT DISTRIBUTED PROPERLY
     .self$name <- 'Differentially private covariance matrix'
-    .self$mechanism <- mechanism
-    .self$varType <- varType
-    .self$n <- checkNValidity(n)
-    .self$epsilon <- epsilon
-    .self$delta <- delta
-    .self$rng <- rng
+    .self$mechanism <- checkMechanism(mechanism, "mechanismLaplace")
+    .self$varType <- checkVariableType(varType, c('integer', 'double', 'numeric', 'logical'))  #NEED TO CHANGE TO ALLOW MULTIPLE VARTYPES  
+    .self$n <- checkN(n)  #NEED TO ADD THING SAYING HOW THIS IS N PER COLUMN
```

Comment:
could this just be a change where we check n per column within the covariance statistic (e.g. looping through the columns), or would the checkN() function need to be changed

Reply:
oh, it only actually needs to be one input value for everything since all columns in data frame should be same size (at lease, I've been assuming that, though I suppose we could have that not be true, hmm). my comment was just to remind myself to add a comment in the covariance statistic to make it clear that by n I mean column length not, say, total number of values across all of the columns you want to find covariance of.